test/test_rng.py

```diff
@@ -348,6 +348,99 @@ def randn_kernel_3d(x: torch.Tensor) -> torch.Tensor:
                 f"Slice {b_idx} std {slice_std} is not well distributed",
             )
 
+    def test_hl_rand_1d(self):
+        @helion.kernel
+        def rand_kernel_tiled_1d(x: torch.Tensor, seed: int) -> torch.Tensor:
+            output = torch.zeros_like(x)
+            (m,) = x.shape
+            for (tile_m,) in hl.tile([m]):
+                output[tile_m] = hl.rand([tile_m], seed=seed)
+            return output
+
+        x_small = torch.ones(128, device=DEVICE)
+        _, output = code_and_output(rand_kernel_tiled_1d, (x_small, 42))
+        _, output2 = code_and_output(rand_kernel_tiled_1d, (x_small, 1337))
+
+        self.assertFalse(
+            torch.allclose(output, output2),
+            "Different seeds should produce different outputs",
+        )
+
+        _, output3 = code_and_output(rand_kernel_tiled_1d, (x_small, 42))
+        self.assertTrue(
+            torch.allclose(output, output3),
+            "Same seed should produce identical outputs",
+        )
+
+        # Check that all values are in [0, 1) range
+        self.assertTrue(torch.all(output >= 0.0), "All values should be >= 0")
+        self.assertTrue(torch.all(output < 1.0), "All values should be < 1")
+
+    def test_hl_rand_2d(self):
+        @helion.kernel
+        def rand_kernel_tiled_2d(x: torch.Tensor, seed: int) -> torch.Tensor:
+            output = torch.zeros_like(x)
+            m, n = x.shape
+            for tile_m, tile_n in hl.tile([m, n]):
+                output[tile_m, tile_n] = hl.rand([tile_m, tile_n], seed=seed)
+            return output
+
+        x_small = torch.ones(128, 128, device=DEVICE)
+        _, output = code_and_output(rand_kernel_tiled_2d, (x_small, 42))
+        _, output2 = code_and_output(rand_kernel_tiled_2d, (x_small, 1337))
+
+        self.assertFalse(
+            torch.allclose(output, output2),
+            "Different seeds should produce different outputs",
+        )
+
+        _, output3 = code_and_output(rand_kernel_tiled_2d, (x_small, 42))
+        self.assertTrue(
+            torch.allclose(output, output3),
+            "Same seed should produce identical outputs",
+        )
+
+        self.assertTrue(torch.all(output >= 0.0), "All values should be >= 0")
+        self.assertTrue(torch.all(output < 1.0), "All values should be < 1")
+
+    def test_hl_rand_3d(self):
+        import helion
+
+        @helion.kernel(ref_mode=helion.RefMode.EAGER)
```

Comment:
Maybe remove the ref_mode setting? This will explicitly test ref eager mode instead of normal Helion compile mode.